### DISCUSSION TITLE
Feature Request: auto-prefixing `unnest(recursive := true)` column names with parent keys?

### DISCUSSION THREAD

**Discussion by cmdlineluser** [duckdb/duckdb#11012] [Ideas]
Recursive unnest is super useful - but it seems there is no way to retain the "parent keys" as part of the resulting column names:

```python
duckdb.sql("""
select unnest(
   {'foo': {'bar': {'x': 1, 'y': 10}}}, 
   recursive := true
)
""")

# ┌───────┬───────┐
# │   x   │   y   │
# │ int32 │ int32 │
# ├───────┼───────┤
# │     1 │    10 │
# └───────┴───────┘
```

i.e. `foo.bar.x` and `foo.bar.y` in this case.

Similar to how [`pandas.json_normalize()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.json_normalize.html) behaves:

```python
pd.json_normalize({'foo': {'bar': {'x': 1, 'y': 10}}})
#    foo.bar.x  foo.bar.y
# 0          1         10
```

Thanks all <3

**Comment by abelcha** [comment]
up

**Comment by noahscantron** [comment]
Please!

**Comment by barbaradlf** [comment]
Same!

**Comment by franolic** [comment]
Pretty please!

**Comment by soerenwolfers** [comment]
Implemented in https://github.com/duckdb/duckdb/pull/19310
